LearningMissionSimulation/PlaygroundGavril.cs

```diff
@@ -206,7 +213,11 @@ public void CreateClassrooms(int classroomCount)
                 {
                     Guid moduleId = moduleIdList[AttributeGenerator.random.Next(0, moduleIdList.Count)];
                     Module module;
-                    moduleDictionary.TryGetValue(moduleId, out module);
+                    var moduleFound = moduleDictionary.TryGetValue(moduleId, out module);
+                    if (! moduleFound  )
+                    {
+                        ReportError(missingResource: "Modules", failedAction: action);
+                    }
```

Comment:
You need an **else** block to handle the case when moduleDictionary.TryGetValue(moduleId, out module) is true.

Reply:
Done

Reply:
This code does not make any sense:
                     if (!moduleDictionary.TryGetValue(moduleId, out module))
                    {
                        ReportError(missingResource: "Modules", failedAction: action);
                    }
                    else 
                    {
                        ReportError(missingResource: "Modules", failedAction: action);
                    }

You report errors in both cases. Which is wrong.  If moduleDictionary.TryGetValue(moduleId, out module) is true then the module is found and you should use it.

Reply:
DONE

Reply:
DONE

Reply:
DONE